### ISSUE TITLE
NVDA does not announce the placeholder text of QuickPick components

### ISSUE THREAD

**Issue by pranshu512** [microsoft/vscode#104664]
I'm validating the accessibility of an extension that uses a QuickPick. However, when I'm navigating using tab key on the QuickPick element, NVDA does not announce the text defined as placeholder but instead: "Type to narrow down results".

NVDA should announce the placeholder text as this is what appears visually.
**Similar issue also observed with JAWS.**
[NVDA issue.zip](https://github.com/microsoft/vscode/files/5075384/NVDA.issue.zip)

**Comment by isidorn** [comment]
I can not reproduce this with buil-in quick pick commands. Ctrl + e or F1 the placeholder gets nicely read.

For extension quick picks the issue seems to be that "Type to narrow down results" is always prepanded to the placeholder message that is read out.
Thus I believe for the quick pick we still have an element with the default aria label, there is another element which has an appropriate aria label.
@chrmarti does this make sense? If you do not have time to fix this this milestone let me know and I can also investigate more. Thanks!

**Comment by chrmarti** [comment]
For showQuickPick() we already set it to the placeholder:
https://github.com/microsoft/vscode/blob/908af6a089c23c61c21dcb3094ef801f76e29ef4/src/vs/base/parts/quickinput/browser/quickInput.ts#L1388

**Comment by isidorn** [comment]
We should maybe look into the dom structure and double check the `aria-labels` and `title` attributes.
I noticed that stablePeek setting is not repsected by the extension quick pick. What is the easiest way that it does not close on blur so I can use dev tools to HTML inspect

**Comment by chrmarti** [comment]
There is a setting:
![image](https://user-images.githubusercontent.com/9205389/90477789-4e7f5600-e12c-11ea-95a4-0caf5ee2f49e.png)

Thanks for looking into this!

**Comment by isidorn** [comment]
@chrmarti I investigated a bit into this and the issue is that aria-label is inccorectly set
Note how the `aria-label` should be the same as `title`.
`aria-label` has the incorrect default value of "Type to narrow down results"
Very easy repro steps is just to open any quick pick, for example turn settings sync on.

Can you maybe look into how the aria-label gets assigned on the input element

![Screenshot 2020-08-20 at 11 10 12](https://user-images.githubusercontent.com/1926584/90750904-c2109700-e2d5-11ea-8279-137484349178.png)

**Comment by chrmarti** [comment]
VoiceOver reads both the placeholder and the aria-label. Changing the aria-label to the placeholder text will have it read twice. Do we have any guidance on how to use these together?

**Comment by isidorn** [comment]
@chrmarti I would just set them to the same value and it is up to the screen readers to optimise this and de-dupe.
@joanmarie do you agree with my statment in general? I believe Orca has smartness to de-dupe, however here we hafve a `placeholder` and `aria-label` set to the same thing.

**Comment by pranshu512** [comment]
@chrmarti @isidorn I found that this issue still repro while using NVDA.

**Comment by isidorn** [comment]
@Pranshu258 There has not yet been an insiders with the fix. It should soon be out in the next 2 hours. Then you can try it out. Thanks

**Comment by joanmarie** [comment]
@isidorn: Orca does do some de-duping, but it doesn't seem to be doing so for placeholder text (yet).

**Comment by joanmarie** [comment]
Ok, Orca master and the gnome-3-38 branch ignore placeholder text when it's the same as the name.

**Comment by isidorn** [comment]
Lovely. Thanks!

**Comment by pranshu512** [comment]
@chrmarti Please reactivate this bug as according to the process only the trusted tester can close this bug after verifying fix on environment.

**Comment by isidorn** [comment]
@pranshu512 first of all welcome to our repository and thanks for creating issues. We are closing issues when they get fixed. If you believe it is not fixed you can let us know and we will re-investigate.
More about our issue process can be found here https://github.com/microsoft/vscode/wiki/Issue-Tracking

**Comment by pranshu512** [comment]
@isidorn @chrmarti  I checked this bug on VS code following version:
 Version: 1.48.2 (user setup)
Commit: a0479759d6e9ea56afa657e454193f72aef85bd0
Date: 2020-08-25T10:13:11.295Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041
 and found that NVDA still does not announce the placeholder text of QuickPick components when i am navigating using keyboard tab key.
NVDA is announcing "quick input list"

**Comment by isidorn** [comment]
@Pranshu258 please use vscode insiders as I have mentioned above https://code.visualstudio.com/insiders/
Thank you!
